packages/react/src/publication/adapters/useReactionController.ts

```diff
@@ -0,0 +1,26 @@
+import { Reaction, ReactionRequest } from '@lens-protocol/domain/use-cases/publications';
+
+import { useSharedDependencies } from '../../shared';
+import { ReactionGateway } from './ReactionGateway';
+import { ReactionPresenter } from './ReactionPresenter';
+
+export function useReactionController() {
+  const { apolloClient } = useSharedDependencies();
+
+  const presenter = new ReactionPresenter(apolloClient.cache);
+  const gateway = new ReactionGateway(apolloClient);
+  const reaction = new Reaction(gateway, presenter);
```

Comment:
This approach creates new adapters at every render cycle which by React design can happen any time.

Although it's not wrong per-se implies that the controller knows that these adapters are probably stateless or the only state they care about is the dependencies provided.

I feel this approach could be error prone as someone might make a change to one of these adapters, for example adding an internal state (think of an in-memory cache for a gateway, or a reference to some UI state for a presenter) and because such state gets replaced with a new one at every re-render they might introduce a malfunction or even a memory leak depending if the such state is able to keep a reference to the adapters n-th instance.

I am not advocating for one over the other but 2 different approaches might be better:
- memoize the creation of these adapters (this could also inform the reader if their state is tied to the hook lifecycle)
- create instances of these adapters as part of the `add` and `remove` methods. This could be helpful to communicate to the reader the presence of an ephemeral state that could be tied to the operation lifecycle (add or remove a reaction).

This last option could be helpful to address your very valid comment here: https://github.com/lens-protocol/lens-sdk/pull/56#discussion_r1060072482

Maybe we can bounce some ideas on this topic.

Reply:
I understand it now, my bad. I saw the 2nd approach in other Controllers (create instances of these adapters as part of the add and remove methods) so I will just do the same.